tests/functional/test_integration_base_class.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+from django.core.exceptions import ImproperlyConfigured
+
+
+def test_missing_identifier(monkeypatch):
+    """
+    Tests that an exception is raised when identifier is missing.
+    """
+    from django_guid.integrations.sentry import SentryIntegration
+
+    monkeypatch.setattr(SentryIntegration, 'identifier', None)
+    with pytest.raises(ImproperlyConfigured, match='`identifier` cannot be None'):
+        SentryIntegration()
+
+
+def test_missing_run_method(monkeypatch, client):
+    """
+    Tests that an exception is raised when the run method has not been defined.
+    """
+    from django_guid.config import settings as guid_settings
+    from django_guid.integrations.sentry import SentryIntegration
+
+    monkeypatch.delattr(SentryIntegration, 'run')
+    monkeypatch.setattr(guid_settings, 'INTEGRATIONS', [SentryIntegration()])
+    with pytest.raises(ImproperlyConfigured, match='The integration `SentryIntegration` is missing a `run` method'):
+        client.get('/api')
+
+
+def test_run_method_not_accepting_kwargs(monkeypatch, client):
+    """
+    Tests that an exception is raised when the run method doesn't accept kwargs.
+    """
+    from django_guid.integrations.sentry import SentryIntegration
+    from django.conf import settings
+    from django_guid.config import Settings
+
+    class BadIntegration(SentryIntegration):
+        def run(self, guid):
+            pass
+
+    monkeypatch.setattr(settings, 'DJANGO_GUID', {'INTEGRATIONS': [BadIntegration()]})
+    with pytest.raises(ImproperlyConfigured, match='Integration method `run` must accept keyword arguments '):
+        Settings()
+
+
+def test_non_callable_methods(monkeypatch, client):
+    """
+    Tests that an exception is raised when any of the integration base methods are non-callable.
+    """
+    from django_guid.integrations.sentry import SentryIntegration
+    from django.conf import settings
+    from django_guid.config import Settings
+
+    x = SentryIntegration()
+
+    # Non-callable run method
+    x.run = 'test'
+    monkeypatch.setattr(settings, 'DJANGO_GUID', {'INTEGRATIONS': [x]})
+    with pytest.raises(ImproperlyConfigured, match='Integration method `run` needs to be made callable for `SentryIntegration`.'):
+        Settings()
+
+    # # Non-callable validate method
+    x.validate = 'test'
+    monkeypatch.setattr(settings, 'DJANGO_GUID', {'INTEGRATIONS': [x]})
+    with pytest.raises(
+        ImproperlyConfigured, match='Integration method `validate` needs to be made callable for `SentryIntegration`.'
+    ):
+        Settings()
+
+    # # Non-callable setup method
+    x.setup = 'test'
+    monkeypatch.setattr(settings, 'DJANGO_GUID', {'INTEGRATIONS': [x]})
+    with pytest.raises(
+        ImproperlyConfigured, match='Integration method `setup` needs to be made callable for `SentryIntegration`.'
+    ):
+        Settings()
+
+
+def test_base():
```

Comment:
This is a function calling a class, not a test asserting anything.

Reply:
it will still fail if you manage to break the base class